edx_solutions_api_integration/management/commands/add_test_poll_responses.py

```diff
@@ -0,0 +1,185 @@
+        """
+        Generates a random answers for a specified user and block
+
+        The currently supported blocks are poll and survey
+        """
+        location = block.location
+        answer = {}
+
+        if block.category == 'poll':
+            # Possible answers come in this format:
+            # ('G', {'img': None, 'img_alt': None, 'label': 'Green'})
+            # and we only need the 'G' bit for saving the answer
+            possible_answers = [a[0] for a in block.answers]
+            # Answer format for StudentModule model:
+            # {"submissions_count": 1, "choice": "R"}
+            answer = {
+                "submissions_count": 1,
+                "choice": random.choice(possible_answers),
+            }
+        elif block.category == 'survey':
+            # Questions come in this format:
+            # ('enjoy', {'img': None, 'img_alt': None, 'label': 'Are you enjoying the course?'})
+            # and we only need the 'enjoy' part
+            questions = [q[0] for q in block.questions]
+            # This happens similarly with answers
+            possible_answers = [a[0] for a in block.answers]
+
+            # Answer format for StudentModule model:
+            # {"submissions_count": 1, "choices": {"enjoy": "Y", "learn": "Y", "recommend": "N"}}
+            answer = {
+                "submissions_count": 1,
+                "choices": {key: random.choice(possible_answers) for key in questions},
+            }
+
+        return StudentModule(
+            module_type='problem',
+            module_state_key=location,
+            student=student,
+            course_id=course_key,
+            state=answer,
+        )
+
+    def handle(self, *args, **options):
+        """
+        Retrieves all XBlocks for a course and generates dummy reponses for
+        all users limited to --limit.
+
+        To use this command to generate answers for polls and suveys, you'll
+        need to create a course, add polls/surveys and users to it.
+        """
+        if not options.get('course_id'):
+            raise CommandError("add_test_poll_responses command requires the parameter --course-id")
+
+        store = modulestore()
+        course_key = CourseKey.from_string(options['course_id'])
+
+        # Get data from store and models
+        courses = store.get_course(course_key)
+        blocks = self.get_course_blocks(store, course_key, ['poll', 'survey'])
+        students = self.get_enrolled_students(course_key)
+
+        # Create product between users and polls
+        combinations = self.generate_combinations(students, blocks, course_key)
+
+        submissions = []
+        # Zip through items, limiting range with num_responses
+        for index, [student, block] in zip(range(options['num_responses']), combinations):
+            submissions.append(self.generate_dummy_submission(student, block, course_key))
+
+        if submissions:
+            StudentModule.objects.bulk_create(submissions)
```

Comment:
Are there any post-save signals for StudentModule? If so perhaps this is an optimisation we cannot make.

Reply:
Also, if we are creating say 50,000 responses, would this work? Perhaps it is better to batch this?

Reply:
@xitij2000 The only model that is also updated is the `XBlock summary` but it isn't needed for the report generation.
I have the resources to update that too, but it might make the code a little more messy.
Should I work on it?

Reply:
That model is for fields that go across users, i.e. a vote tally for a voting block, or the number of responses in this block. Have you tested what happens if you don't create that block? If it doesn't break the report generation then I'd say leave it, we're not using that data anyway. If it is necessary, then of course you should add it.

Reply:
@xitij2000 That doesn't break the report generation, just messes up the visualization on LMS and Apros if you log in as a normal user.